test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/Parallel/ParallelProxyDiscoveryManagerTests.cs

```diff
@@ -147,7 +147,7 @@ public void DiscoverTestsShouldProcessAllSources()
     }
 
     [TestMethod]
-    public void HandlePartialDiscoveryCompleteShouldReturnTrueIfDiscoveryWasAbortedWithEventHandler()
+    public void HandlePartialDiscoveryCompleteShouldReturnTrueIfDiscoveryWasAbortedBeforeBeingStartedWithEventHandler()
```

Comment:
This test calls `Abort` without having called `DiscoverTests` before, that seems illogical but nothing in our code prevents this state so I decided to keep the test for now.

As we were initializing to -1 the number of sources and never calling `DiscoverTests` which resets the value to the actual number of sources (i.e. 0+) we were having assertion failures.

Reply:
Cool, I planned to investigate why I am seeing the failure on -1 sources, I thought it is something like this where the initial step is skipped in a unit test. So now I don't have to.